sky/skylet/providers/aws/node_provider.py

```diff
@@ -662,3 +662,17 @@ def fillout_available_node_types_resources(
                     + "."
                 )
         return cluster_config
+
+
+class AWSNodeProviderV2(AWSNodeProvider):
+    # The new version of the AWS node provider uses a new IAM role
+    # with different permissions against original ray-autoscaler-v1
+    # for both the head node and worker nodes to support AWS SSO login
+    # system (see #1489).
+    # We did not overwrite the original AWSNodeProvider class to avoid
+    # breaking existing clusters. Otherwise, the existing clusters will
+    # have a new launch_hash and will have new node(s) launched, causing
+    # the existing nodes to leak.
```

Comment:
```suggestion
class AWSNodeProviderV2(AWSNodeProvider):
    """Same as V1, except head and workers use a SkyPilot IAM role.

    The new version of the AWS node provider supports AWS SSO 
    (see #1489), by using a new IAM role with different permissions 
    than the original ray-autoscaler-v1 for both the head node and 
    worker nodes.

    We did not overwrite the original AWSNodeProvider class to avoid
    breaking existing clusters. Otherwise, the existing clusters will
    have a new launch_hash and will have new node(s) launched, causing
    the existing nodes to leak.
```